pkg/console/controllers/managedclusters/controller.go

```diff
@@ -197,20 +213,143 @@ func (c *ManagedClusterController) ValidateManagedClusterClientConfigs(ctx conte
 			continue
 		}
 
-		validatedClientConfigs[clusterName] = &clientConfigs[0]
+		valid = append(valid, &managedCluster)
+	}
+
+	return valid, nil, ""
+}
+
+func (c *ManagedClusterController) SyncOAuthClientManagedClusterViews(ctx context.Context, operatorConfig *operatorv1.Console, managedClusters []*clusterv1.ManagedCluster) ([]*unstructured.Unstructured, error, string) {
+	errs := []string{}
+	mcvs := []*unstructured.Unstructured{}
+	for _, managedCluster := range managedClusters {
+		required := managedclusterviewsub.DefaultOAuthClientView(managedCluster.Name)
+		mcv, err := c.dynamicClient.Resource(api.ManagedClusterViewGroupVersionResource).Namespace(managedCluster.Name).Create(ctx, required, metav1.CreateOptions{})
+		if err != nil && apierrors.IsAlreadyExists(err) {
+			mcvName, _ := managedclusterviewsub.GetName(required)
+			mcv, err = c.dynamicClient.Resource(api.ManagedClusterViewGroupVersionResource).Namespace(managedCluster.Name).Get(ctx, mcvName, metav1.GetOptions{})
+		}
+
+		if err != nil || mcv == nil {
+			errs = append(errs, fmt.Sprintf("Error syncing managed cluster view for oauth client for cluster %s: %v", managedCluster.Name, err))
+		} else {
+			mcvs = append(mcvs, mcv)
+		}
+	}
+
+	if len(errs) > 0 {
+		return nil, errors.New(strings.Join(errs, "\n")), "ManagedClusterViewOAuthClientSyncError"
+	}
+
+	return mcvs, nil, ""
+}
+
+func (c *ManagedClusterController) SyncOAuthClientCreationManagedClusterActions(ctx context.Context, operatorConfig *operatorv1.Console, localOAuthClient *oauthv1.OAuthClient, oAuthClientMCVs []*unstructured.Unstructured) ([]*unstructured.Unstructured, error, string) {
+	managedClusterList := []string{}
+	managedClusterListErrors := []string{}
+	for _, managedClusterOAuthView := range oAuthClientMCVs {
+		status, statusErr := managedclusterviewsub.GetStatus(managedClusterOAuthView)
+		if !status || statusErr != nil {
+			namespace, namespaceErr := managedclusterviewsub.GetNamespace(managedClusterOAuthView)
+			if namespaceErr != nil || namespace == "" {
+				managedClusterListErrors = append(managedClusterListErrors, fmt.Sprintf("Unable to create oauth client for cluster %v: managed cluster view status unknown", namespace))
+			} else {
+				managedClusterList = append(managedClusterList, namespace)
+			}
+		}
+	}
+
+	errs := []string{}
+	mcas := []*unstructured.Unstructured{}
+	secretString := oauthsub.GetSecretString(localOAuthClient)
```

Comment:
Ideally we'd have different secrets for each OAuthClient, but OK for a tech preview.